x-pack/plugins/siem/public/components/page/hosts/authentications_table/index.tsx

```diff
@@ -303,3 +305,24 @@ const getAuthenticationColumns = (): [
       }),
   },
 ];
+
+const getAuthenticationColumnsCurated = (pageType: hostsModel.HostsType) => {
```

Comment:
I think that you can avoid the map and redefine the exclude array each time by doing that 
```
// Array of column names to exclude from host details pages
const columnsToExcludedForDetailsPage = [
  i18n.LAST_FAILED_DESTINATION,
  i18n.LAST_SUCCESSFUL_DESTINATION,
];

const getAuthenticationColumnsCurated = (pageType: hostsModel.HostsType) => {
  const columnsCurated = getAuthenticationColumns(pageType);

  if (pageType === 'details') {
    // Remove excluded columns from array
    for (let i = columnsCurated.length - 1; i >= 0; i--) {
      const columnObject = columnsCurated[i];
      const colName: string = columnObject.name != null ? columnObject.name.toString() : '';
      if (columnsToExcludedForDetailsPage.includes(colName)) {
        columnsCurated.splice(i, 1);
      }
    }
  }

  return columnsCurated;
};
```

and I will add these tests
```
describe('columns', () => {
    test('On the main page, We expect to get all the columns', () => {
      expect(getAuthenticationColumnsCurated(hostsModel.HostsType.page).length).toEqual(9);
    });

    test('On Details page, We expect to remove two columns', () => {
      const columns = getAuthenticationColumnsCurated(hostsModel.HostsType.details);
      expect(columns.length).toEqual(7);
    });

    test('On details page, we should not have Last Failed Destination column', () => {
      const columns = getAuthenticationColumnsCurated(hostsModel.HostsType.details);
      expect(
        columns.includes(
          (col: Columns<AuthenticationsEdges>) => col.name === i18n.LAST_FAILED_DESTINATION
        )
      ).toEqual(false);
    });

    test('On details page, we should not have Last Successful Destination column', () => {
      const columns = getAuthenticationColumnsCurated(hostsModel.HostsType.details);
      expect(
        columns.includes(
          (col: Columns<AuthenticationsEdges>) => col.name === i18n.LAST_SUCCESSFUL_DESTINATION
        )
      ).toEqual(false);
    });
  });
```

to make the test run you will have to chnage the imports to that
```
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { getOr } from 'lodash/fp';
import * as React from 'react';
import { Provider as ReduxStoreProvider } from 'react-redux';

import { apolloClientObservable, mockGlobalState } from '../../../../mock';
import { AuthenticationsEdges } from '../../../../graphql/types';
import { createStore, hostsModel, State } from '../../../../store';
import { Columns } from '../../../load_more_table';

import { mockData } from './mock';
import * as i18n from './translations';
import { AuthenticationTable, getAuthenticationColumnsCurated } from '.';
```

Reply:
Thanks, @XavierM. For the function cleanup, I already implemented @stephmilovic's recommended simplification. Should I stick with that or use yours?

As for the tests, I just did another table render test (shown below), but with the page set to the details page `type`. Is that a sufficient test, or would you prefer the more granular testing you are recommending above?

```
  describe('rendering', () => {
    test('it renders the hosts page events table', () => {
      const wrapper = shallow(
        <ReduxStoreProvider store={store}>
          <EventsTable
            loading={false}
            data={mockData.Events.edges.map(i => i.node)}
            totalCount={mockData.Events.totalCount}
            tiebreaker={getOr(null, 'endCursor.tiebreaker', mockData.Events.pageInfo)!}
            hasNextPage={getOr(false, 'hasNextPage', mockData.Events.pageInfo)!}
            nextCursor={getOr(null, 'endCursor.value', mockData.Events.pageInfo)}
            loadMore={loadMore}
            type={hostsModel.HostsType.page}
          />
        </ReduxStoreProvider>
      );

      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('it renders the host details page events table', () => {
      const wrapper = shallow(
        <ReduxStoreProvider store={store}>
          <EventsTable
            loading={false}
            data={mockData.Events.edges.map(i => i.node)}
            totalCount={mockData.Events.totalCount}
            tiebreaker={getOr(null, 'endCursor.tiebreaker', mockData.Events.pageInfo)!}
            hasNextPage={getOr(false, 'hasNextPage', mockData.Events.pageInfo)!}
            nextCursor={getOr(null, 'endCursor.value', mockData.Events.pageInfo)}
            loadMore={loadMore}
            type={hostsModel.HostsType.details}
          />
        </ReduxStoreProvider>
      );

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
```

Reply:
I do not think that it is working because snapshots are not showing the table rendering, I think that you will have to do granular tests

Reply:
👍